test/SlowTests/Client/TimeSeries/Query/TimeSeriesJavascriptProjections.cs

```diff
@@ -0,0 +1,2008 @@
+
+            public TimeSeriesRawResult Stocks { get; set; }
+        }
+
+        private class CustomJsFunctionResult
+        {
+            public double Max { get; set; }
+
+            public bool HasApple { get; set; }
+
+            public List<double> Accuracies { get; set; }
+        }
+
+        private class CustomJsFunctionResult2
+        {
+            public double TotalMax { get; set; }
+
+            public double TotalMin { get; set; }
+
+            public double AvgOfAvg { get; set; }
+
+            public double MaxGroupSize { get; set;  }
+        }
+
+
+        [Fact]
+        public void CanCallTimeSeriesDeclaredFunctionFromJavascriptProjection()
+        {
+            using (var store = GetDocumentStore())
+            {
+                var baseline = DateTime.Today;
+
+                using (var session = store.OpenSession())
+                {
+                    session.Store(new Person
+                    {
+                        Name = "Oren",
+                        Age = 60,
+                        WorksAt = "companies/1"
+                    }, "people/1");
+
+                    session.Store(new Company
+                    {
+                        Name = "HR"
+                    }, "companies/1");
+
+                    var tsf = session.TimeSeriesFor("people/1");
+
+                    tsf.Append("Stocks", baseline.AddMinutes(61), "tags/1", new[] { 59d });
+                    tsf.Append("Stocks", baseline.AddMinutes(62), "tags/2", new[] { 79d });
+                    tsf.Append("Stocks", baseline.AddMinutes(63), "tags/2", new[] { 69d });
+
+                    tsf.Append("Stocks", baseline.AddMonths(1).AddMinutes(61), "tags/1", new[] { 159d });
+                    tsf.Append("Stocks", baseline.AddMonths(1).AddMinutes(62), "tags/1", new[] { 179d });
+                    tsf.Append("Stocks", baseline.AddMonths(1).AddMinutes(63), "tags/2", new[] { 169d });
+
+                    tsf = session.TimeSeriesFor("companies/1");
+
+                    tsf.Append("Stocks", baseline.AddMinutes(61), "tags/1", new[] { 559d });
+                    tsf.Append("Stocks", baseline.AddMinutes(62), "tags/2", new[] { 579d });
+                    tsf.Append("Stocks", baseline.AddMinutes(63), "tags/2", new[] { 569d });
+
+                    tsf.Append("Stocks", baseline.AddMonths(1).AddMinutes(61), "tags/1", new[] { 659d });
+                    tsf.Append("Stocks", baseline.AddMonths(1).AddMinutes(62), "tags/1", new[] { 679d });
+                    tsf.Append("Stocks", baseline.AddMonths(1).AddMinutes(63), "tags/2", new[] { 669d });
+
+                    session.SaveChanges();
+                }
+
+                using (var session = store.OpenSession())
+                {
+                    var rawQuery = session.Advanced.RawQuery<CustomRawQueryResult3>(
+@"declare timeseries out(d) 
+{
+    from d.Stocks between $start and $end
+    where Tag != 'tags/2'
```

Comment:
How can I pass this as a param?
So for `Series`, it will be `tags/2` but for `Series2` it will be `tags/3` ?

Reply:
```
declare timeseries out(d, t) 
{
    from d.Stocks between $start and $end
    where Tag != t
}
from People as p
where p.Age > 49
load p.WorksAt as Company
select {
    Series: out(p, 'tags/2'),
    Series2: out(Company, 'tags/3')
}
```

added test